### ISSUE TITLE
Training Code Issues in DaRF

### ISSUE THREAD

**Issue by UranusITS** [KU-CVLAB/DaRF#4]
Thank you for your outstanding work.

I am reaching out regarding some issues encountered with the training code in your repository. While I greatly appreciate the value of your work, I have faced multiple bugs that hinder the code's execution. These include both syntactic and logical errors, which impede the training process.

For instance, in building the ScanNet training dataset, the `filenames` variable is not returned. This variable is required in `__init__` when `split` is not `"render"`.

https://github.com/KU-CVLAB/DaRF/blob/eb29b7dee58f75e2f973786bb5956227ebbbcd19/plenoxels/datasets/scannet_dataset.py#L279-L298

Additionally, there are bugs in the validation part of your trainer. If I try to fetch data from the test dataset, the type of `idx` becomes an `int` rather than a `list` or a `torch.Tensor`, leading to a reduction in dimensions for variables like `rand_rays_o`, `rand_rays_d`, and others related to `idx`.

https://github.com/KU-CVLAB/DaRF/blob/eb29b7dee58f75e2f973786bb5956227ebbbcd19/plenoxels/datasets/base_dataset.py#L155-L158

Furthermore, I find bugs in the `evaluate_metrics` section. Although I am not fully conversant with this part, I noticed that the shape of `preds["rgb"]` cannot be reshaped into the required image shape.

https://github.com/KU-CVLAB/DaRF/blob/eb29b7dee58f75e2f973786bb5956227ebbbcd19/plenoxels/runners/base_trainer.py#L593-L610

So far, I have only managed to identify these bugs, as troubleshooting them step-by-step has been quite challenging.

Could you please consider updating the code to resolve these issues? Such updates would be immensely helpful in utilizing your research and enhancing its reproducibility.

Thank you for your attention to this matter.

**Comment by hg010303** [comment]
Thank you for thoroughly reviewing our paper and code!

We revisited the errors you pointed out, clone the code from GitHub, and ran it again. For the first error, which seemed to occur in the fewshot part, we have made corrections and updated the code. Thank you for bringing it to our attention.

However, for the second and third errors, when we ran the code again, no issues were encountered. During training and validation, even when using flags like render-only or validate-only separately, no errors were found in my environment.

During testing, **self.patch_rendering** and **self.rand_rays_o** are specified as None, causing the code to enter the else statement below. In such situations, we expect no issues with data type conversion to int or shape-related problems.
https://github.com/KU-CVLAB/DaRF/blob/47b2d1a23d13f0d149e55cf8fd2195ec42093d1e/plenoxels/datasets/base_dataset.py#L161-L249

If you have any further questions or concerns, please feel free to ask!

**Comment by UranusITS** [comment]
Thank you for your response, and I appreciate your insights. I would like to clarify that the last two questions I raised were actually related to challenges I encountered while setting up my own dataset, and I apologize for any confusion.

I also want to express my appreciation for your novel work in the field. If I encounter any further questions or require assistance in the future, I will not hesitate to reach out to you. Thank you for your time.
